### DISCUSSION TITLE
xsns_53_sml.ino Logarex Smart Meter

### DISCUSSION THREAD

**Discussion by wb86** [arendst/Tasmota#11741] [Q&A] [answered]
Hello,

I have a LK13BE803039 smart meter from Loagrex.
Its an Obis speaking meter, so I used this script:

https://tasmota.github.io/docs/Smart-Meter-Interface/#logarex-lk13be-obis

`+1,12,o,0,9600,LK13BE,13,30,2F3F210D0A,063035310D0A`

When I try "sensor53 d1" I got a verry crypted message like:

`19:10:40.132 : KS#__knSCs;sS+C#S#__knSCs;s#S++C++S#__knSCs;s+S++C;S#_oL,kSCs;s3S++C#S#__knS#s;sS++G+sSC`

But when I change the script to:

`+1,12,m,0,9600,LK13BE,13,30,2F3F210D0A,063035310D0A`

I got this message:

```
19:03:04.815 : 2f 4c 4f 47 35 4c 4b 31 33 42 45 38 30 33 30 33 => /LOG5LK13BE80303
19:03:04.851 : 39 0d 0a 0d 0a 31 2d 30 3a 39 36 2e 31 2e 30 2a => 9    1-0:96.1.0*
19:03:04.885 : 32 35 35 28 30 30 31 4c 4f 47 XX XX XX XX XX XX => 255(001LOGXXXXXX
19:03:04.913 : XX XX XX XX XX 0d 0a 31 2d 30 3a 31 2e 38 2e 30 => XXXX)  1-0:1.8.0
19:03:04.940 : 2a 32 35 35 28 30 31 30 31 34 32 2e 34 32 37 35 => *255(010142.4275
19:03:04.967 : 2a 6b 57 68 29 0d 0a 31 2d 30 3a 32 2e 38 2e 30 => *kWh)  1-0:2.8.0
19:03:04.996 : 2a 32 35 35 28 30 30 30 30 30 32 2e 30 35 32 30 => *255(000002.0520
19:03:04.023 : 2a 6b 57 68 29 0d 0a 31 2d 30 3a 31 36 2e 37 2e => *kWh)  1-0:16.7.
19:03:04.051 : 30 2a 32 35 35 28 30 30 30 33 30 39 2a 57 29 0d => 0*255(000309*W) 
19:03:04.079 : 0a 31 2d 30 3a 33 32 2e 37 2e 30 2a 32 35 35 28 =>  1-0:32.7.0*255(
19:03:04.108 : 32 33 32 2e 33 2a 56 29 0d 0a 31 2d 30 3a 35 32 => 232.3*V)  1-0:52
19:03:04.135 : 2e 37 2e 30 2a 32 35 35 28 32 33 31 2e 35 2a 56 => .7.0*255(231.5*V
19:03:04.163 : 29 0d 0a 31 2d 30 3a 37 32 2e 37 2e 30 2a 32 35 => )  1-0:72.7.0*25
19:03:04.191 : 35 28 32 33 33 2e 32 2a 56 29 0d 0a 31 2d 30 3a => 5(233.2*V)  1-0:

```
How can I transfer this results to the main page?

```
1,1-0:1.8.0*255(@1,Gesamtverbrauch,KWh,total,4
1,1-0:1.8.0*96(@1,Verbrauch 1 Tag,KWh,total_1d,4
1,1-0:1.8.0*97(@1,Verbrauch 7 Tage,KWh,total_7d,4
1,1-0:1.8.0*98(@1,Verbrauch 30 Tage,KWh,total_30d,4
1,1-0:1.8.0*99(@1,Verbrauch 365 Tage,KWh,total_365d,4
1,1-0:16.7.0*255(@1,Verbrauch aktuell,W,current,20
```

Does not work for me.

Thank You.

Tim

**Comment by gemu2015** [comment]
this is a very strange finding! 
the serial port in OBIS is 7E1 but it actually uses 8N1 and just ignores the parity.
in Modbus the serial port is set to 8N1 and should not differ from OBIS since bit 7 should be zero anyway.
you can not decode OBIS with the modbus settings because Modbus has a checksum that is not present in OBIS.

maybe an interference between transmitter and receiver ?
can you please check this descriptor ? (no data sending):

+1,12,o,0,9600,LK13BE

and can you please try hardware serial port and recheck ?

**Comment by gemu2015** [comment]
i meant this obis descriptor
+1,12,o,0,9600,LK13BE

**Comment by wb86** [comment, marked as answer]
Hej,

I use a D1 mini with a Hichi IR ttl receiver/transmitter.

With this script "+1,12,o,0,9600,LK13BE" nothing happened.
Maybe the meter needs a start signal?

Because when I use a flashlight to start and mount the receiver quickly I get this in the console:

```
11:17:04.114 
11:17:04.164 : 
11:17:04.214 : 1-0:81.7.26*255(044*deg)
11:17:04.265 : 
11:17:04.315 : 1-0:14.7.0*255(50.0*Hz)
11:17:04.364 : 
11:17:04.415 : 1-0:1.8.0*96(00007.3*
11:17:04.465 : 
11:17:04.514 : 1-0:96.90.2*255(0F66)
11:17:04.565 : 
11:17:04.614 : 1-0:97.97.0*255(00000000)
11:17:04.664 : 
11:17:04.715 : !
11:17:04.764 :
``` 

But only this 0,5 seconds. Afterwards its quiet again. :(

When I switch to "+1,12,m,0,9600,LK13BE,13,30,2F3F210D0A,063035310D0A" its the same I mentioned above.

I think the receiver still works, but there is a mistake in my parameters?

**Reply by gemu2015** [reply]
did you try with hardware serial ? (pins 3 and 1)

i myself cant test your configuration because my 3 meters work with hardware serial and are glued into housings.

but you may easily generate a "blink" signal with the script.

use +1,12,o,0,9600,LK13BE

and in section >B

; this sets pin 13 to output and led off 2. parameter is the character O for Output not 0 (zero)
spinm(13 O)
spin(13 1)

and in section >S

; this generates a blink on pin 13 every 3 seconds
if upsecs%3==0 {
spin(13 0)
delay(10)
spin(13 1)
}

**Comment by wb86** [comment]
Hej, 
I am happy. I tried the pins 3 and 1 and it works.
But why? I thought that every pin can be programmed.
Never mind. It works. ;)
Thank You.

**Reply by gemu2015** [reply]
software serial blocks while transmitting and is only supported for read only or half duplex.
Anyhow it that should be mentioned in the docs. will add that info asap.
